tardis/energy_input/tests/test_gamma_ray_channel.py

```diff
@@ -0,0 +1,192 @@
+    )
+
+    return config_reader.Configuration.from_yaml(yml_path)
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_simulation_state(gamma_ray_config, atomic_dataset):
+    """
+    Parameters
+    ----------
+    gamma_ray_config: Tardis configuration
+    atomic_dataset: Tardis atomic-nuclear dataset
+
+    Returns
+    -------
+    Tardis simulation state
+    """
+
+    gamma_ray_config.model.structure.velocity.start = 1.0 * u.km / u.s
+    gamma_ray_config.model.structure.density.rho_0 = 5.0e2 * u.g / u.cm**3
+    gamma_ray_config.supernova.time_explosion = 150 * u.d
+
+    return SimulationState.from_config(
+        gamma_ray_config, atom_data=atomic_dataset
+    )
+
+
+def test_calculate_cell_masses(gamma_ray_simulation_state):
+    """Function to test calculation of shell masses.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    """
+    volume = 2.70936170e39 * u.cm**3
+    density = 5.24801665e-09 * u.g / u.cm**3
+    desired = volume * density
+
+    shell_masses = gamma_ray_simulation_state.composition.calculate_cell_masses(
+        gamma_ray_simulation_state.geometry.volume
+    )
+
+    npt.assert_allclose(shell_masses[0], desired)
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56", "Fe-52", "Cr-48"])
+def test_isotope_dicts(gamma_ray_simulation_state, nuclide_name):
+    """
+    Function to test if the right names for the isotopes are present as dictionary keys.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+    nuclide = rd.Nuclide(nuclide_name)
+    raw_isotope_abundance = (
+        gamma_ray_simulation_state.composition.raw_isotope_abundance
+    )
+    composition = gamma_ray_simulation_state.composition
+    cell_masses = composition.calculate_cell_masses(
+        gamma_ray_simulation_state.geometry.volume
+    )
+    iso_dict = create_isotope_dicts(raw_isotope_abundance, cell_masses)
+
+    for isotope_dict in iso_dict.values():
+        assert nuclide_name.replace("-", "") in isotope_dict.keys()
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56", "Fe-52", "Cr-48"])
+def test_inventories_dict(gamma_ray_simulation_state, nuclide_name):
+    """
+    Function to test if the inventories dictionary is created correctly.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+
+    nuclide = rd.Nuclide(nuclide_name)
+    raw_isotope_abundance = (
+        gamma_ray_simulation_state.composition.raw_isotope_abundance
+    )
+    composition = gamma_ray_simulation_state.composition
+    cell_masses = composition.calculate_cell_masses(
+        gamma_ray_simulation_state.geometry.volume
+    )
+
+    iso_dict = create_isotope_dicts(raw_isotope_abundance, cell_masses)
+    inventories_dict = create_inventories_dict(iso_dict)
```

Comment:
This is repeated right? Perhaps move it into a separate fixture?

Reply:
This has been taken care of. Thanks.